smithy-model/src/main/java/software/amazon/smithy/model/validation/NodeValidationVisitor.java

```diff
@@ -388,6 +391,32 @@ public List<ValidationEvent> memberShape(MemberShape shape) {
         return events;
     }
 
+    public List<ValidationEvent> checkNullMember(MemberShape shape) {
+        List<ValidationEvent> events = new ArrayList<>();
+        if (!shape.asMemberShape().filter(nullableIndex::isMemberNullable).isPresent()) {
```

Comment:
Nitpick: `shape` is passed in as `MemberShape` so we don't need the `asMemberShape().filter()` do we? Switching to just `shape` removes any apparent (but not practical) ambiguity between the "member shape is not nullable" and "not a member shape" cases that are logically possible.

Reply:
Ninja'd by Kevin with a better version of this comment!